test/Microsoft.TestPlatform.BlameDataCollector.UnitTests/BlameCollectorTests.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.BlameDataCollector.UnitTests
+{
+    using System;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.DataCollection;
+    using Moq;
+    using Microsoft.TestPlatform.BlameDataCollector;
+    using System.Xml;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+    using System.IO;
+    using System.Collections.Generic;
+
+    [TestClass]
+    public class BlameCollectorTests
+    {
+        private DataCollectionEnvironmentContext context;
+        private DataCollectionContext dataCollectionContext;
+        private BlameCollector blameDataCollector;
+        private Mock<DataCollectionLogger> mockLogger;
+        private Mock<DataCollectionEvents> mockDataColectionEvents;
+        private Mock<DataCollectionSink> mockDataCollectionSink;
+        private Mock<IBlameReaderWriter> mockBlameReaderWriter;
+        private XmlElement configurationElement;
+
+        public BlameCollectorTests()
+        {
+            // Initializing mocks
+            this.mockLogger = new Mock<DataCollectionLogger>();
+            this.mockDataColectionEvents = new Mock<DataCollectionEvents>();
+            this.mockDataCollectionSink = new Mock<DataCollectionSink>();
+            this.mockBlameReaderWriter = new Mock<IBlameReaderWriter>();
+            this.blameDataCollector = new BlameCollector(this.mockBlameReaderWriter.Object);
+
+            // Initializing members
+            TestCase testcase = new TestCase();
+            testcase.Id = Guid.NewGuid();
+            this.dataCollectionContext = new DataCollectionContext(testcase);
+            this.configurationElement = null;
+            this.context = new DataCollectionEnvironmentContext(this.dataCollectionContext);
+        }
+
+        [TestMethod]
+        public void InitializeShouldThrowExceptionIfDataCollectionLoggerIsNull()
+        {
+            Assert.ThrowsException<ArgumentNullException>(() =>
+            {
+                this.blameDataCollector.Initialize(this.configurationElement,
+                    this.mockDataColectionEvents.Object, this.mockDataCollectionSink.Object,
+                    (DataCollectionLogger)null, null);
+            });
+        }
+
+        [TestMethod]
+        public void TriggerTestCaseStartedHandlerShouldIncreaseTestStartCount()
```

Comment:
This test is testing an internal implementation detail. We should care about the user behavior.